### PR TITLE
Designate: recommend deploying DNS in a cluster in HA deployment (SOC-10636)

### PR DESCRIPTION
**Pull request by guangyee**
…10636)

In HA deployment, we should recommend users to deploy both DNS and Designate
into the same cluster nodes to avoid single point of failure.

### CODE REVIEW

File: xml/depl_nodes.xml [hunk 1]
```diff
@@ -171,6 +171,11 @@
     barclamp that is not the admin node. The admin node is not added to the
     public network. So another node is needed that can be attached to the public
     network and appear in the designate default pool.
+
+    In High Availability deployment where Designate services are running in a cluster,
+    it is highly recommended that DNS services are running in the same cluster as well.
+    Therefore, DNS barclamp should be applied the the same nodes as the
```
Comment (nicolasbock) [inline xml/depl_nodes.xml]:
```suggestion
    Therefore, the DNS barclamp should be applied to the same nodes as the
```

Comment (nicolasbock) [reply xml/depl_nodes.xml]: Why is it better to run DNS in the same cluster? I get that a DNS cluster is a good thing (:tm:) but what's the advantage of it being the same cluster?

Comment (guangyee) [reply xml/depl_nodes.xml]: You're right. It doesn't have to be in the same cluster. But we usually group all the related services in the same cluster right? I can change it from "same cluster" to "a cluster". Please let me know.

Comment (nicolasbock) [reply xml/depl_nodes.xml]: Ah ok, that makes sense. Yes, in CI we do, but who is to say what a customer might prefer? :wink: Let's just change it to "a cluster" as you suggest. Then we are definitely on the safe side :smile:

Comment (guangyee) [reply xml/depl_nodes.xml]: done

File: xml/depl_nodes.xml [hunk 2]
```diff
@@ -171,6 +171,12 @@
     barclamp that is not the admin node. The admin node is not added to the
     public network. So another node is needed that can be attached to the public
     network and appear in the designate default pool.
+
+    In High Availability deployment where Designate services are running in a cluster,
```
Comment (nicolasbock) [inline xml/depl_nodes.xml]:
```suggestion
    In High Availability deployments where Designate services are running in a cluster,
```

File: xml/depl_nodes.xml [hunk 3]
```diff
@@ -171,6 +171,12 @@
...
+    it is highly recommended that DNS services are running in a cluster as well.
+    For example, in a typical High Availability deployment where the controllers
+    are deployed in 3-node cluster, DNS barclamp should be applied to all the
```
Comment (nicolasbock) [inline xml/depl_nodes.xml]:
```suggestion
    are deployed in a 3-node cluster, the DNS barclamp should be applied to all the
```

File: xml/depl_nodes.xml (hunk 2 above)
Comment (guangyee) [reply xml/depl_nodes.xml]: done

File: xml/depl_nodes.xml (hunk 3 above)
Comment (guangyee) [reply xml/depl_nodes.xml]: good catch!

**Pull request merged**